agent/engine/task_manager.go

```diff
@@ -148,6 +149,18 @@ func (mtask *managedTask) overseeTask() {
 		if !mtask.GetKnownStatus().Terminal() {
 			// If we aren't terminal and we aren't steady state, we should be able to move some containers along
 			llog.Debug("Task not steady state or terminal; progressing it")
+
+			// TODO: Add new task resources provisioned state ?
+			err := mtask.SetupPlatformResources()
+			if err != nil {
```

Comment:
That still doesn't obviate the need for a test here. You have a code path that's dealing with an error being returned. So, having a test case to cover that is necessary.